src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8610,31 +8612,34 @@ private VariableComponentSyntax ParseDeconstructionComponent(bool topLevel = fal
                     designation = this.AddError(designation, ErrorCode.ERR_TypeExpected);
                 }
 
-                result = _syntaxFactory.TypedVariableComponent(type, designation);
+                result = _syntaxFactory.DeclarationExpression(type, designation);
             }
 
             return
-                topLevel ? (ComponentHasTypes(result) ? CheckFeatureAvailability(result, MessageID.IDS_FeatureTuples) : null) : result;
+                topLevel ? (TypeFoundInDeconstructionDeclarationVariables(result) ? CheckFeatureAvailability(result, MessageID.IDS_FeatureTuples) : null) : result;
         }
 
-        private static bool ComponentHasTypes(VariableComponentSyntax node)
+        /// <summary>
+        /// Checks if we can find at least one type in the deconstruction variables
+        /// </summary>
+        private static bool TypeFoundInDeconstructionDeclarationVariables(ExpressionSyntax node)
         {
             switch (node.Kind)
             {
-                case SyntaxKind.ParenthesizedVariableComponent:
+                case SyntaxKind.TupleExpression:
                     {
-                        var syntax = (ParenthesizedVariableComponentSyntax)node;
-                        if (syntax.Variables.Count <= 1) return false; // don't count 1ples
-                        for (int i = 0; i < syntax.Variables.Count; i++)
+                        var syntax = (TupleExpressionSyntax)node;
+                        if (syntax.Arguments.Count <= 1) return false; // don't count 1ples
+                        for (int i = 0; i < syntax.Arguments.Count; i++)
                         {
-                            if (ComponentHasTypes(syntax.Variables[i])) return true;
+                            if (TypeFoundInDeconstructionDeclarationVariables(syntax.Arguments[i].Expression)) return true;
                         }
 
                         return false;
                     }
-                case SyntaxKind.TypedVariableComponent:
+                case SyntaxKind.DeclarationExpression:
                     {
-                        var syntax = (TypedVariableComponentSyntax)node;
+                        var syntax = (DeclarationExpressionSyntax)node;
                         if (syntax.Type.IsMissing || syntax.Designation.IsMissing) return false;
```

Comment:
Yes, if we try to parse `(x, var y) = e;` as a d-declaration, then the type on `x` will be missing.
Similarly, in `(int, var y) = e;`, the designation will be missing.